### PR TITLE
Bug 2074299: update golang.org/x/crypto to address security vulnerabilities

### PR DESCRIPTION
**Pull request by r4f4**
A broken cryptographic algorithm flaw was found in golang.org/x/crypto/ssh. This issue causes a client to fail authentification with RSA keys to servers that reject signature algorithms based on SHA-2, enabling an attacker to crash the server, resulting in a loss of availability.

### CODE REVIEW

**Comment by r4f4** [comment]
/retitle Bug 2074299: update golang.org/x/crypto to address security vulnerabilities

File: go.mod [hunk 1]
```diff
@@ -247,3 +247,6 @@ replace github.com/openshift/api => github.com/openshift/api v0.0.0-202208231438
 replace github.com/terraform-providers/terraform-provider-nutanix => github.com/nutanix/terraform-provider-nutanix v1.5.0
 
 replace github.com/mattn/go-sqlite3 => github.com/mattn/go-sqlite3 v1.10.0
+
+// https://bugzilla.redhat.com/show_bug.cgi?id=2064702
+replace golang.org/x/crypto => golang.org/x/crypto v0.0.0-20220315160706-3147a52a75dd
```
Comment (patrickdillon) [inline go.mod:252]: This `replace` is needed for any indirect dependencies we have, but the direct dependency we have in the first `require` block is a different, earlier, version than this... just curious did you try bumping that direct dependency?

Long term we should be able to remove this `replace` once all the indirect dependencies no longer use the bad version of the crypto package.

**Review by patrickdillon — COMMENTED** [review]
/lgtm
/skip
none of the test failures look related to these changes

File: go.mod (hunk 1 above)
Comment (r4f4) [reply go.mod:252]: I'm hoping that by only adding the `replace` this PR will be easily cherry-pickable to 4.11, the version against which the CVEs were reported.
I'll be happy to revisit this for 4.12 in a separate PR.

Comment (r4f4) [reply go.mod:255]: @patrickdillon I rebased on top of current master and commit 1dd74ca7a8bcd2978d056bf742e9cff409f5fbd0 took care of bumping the x/crypto package.

**Comment by r4f4** [comment]
/override ci/prow/e2e-aws-ovn
unrelated e2e test failures

**Comment by r4f4** [comment]
/override ci/prow/e2e-aws-ovn ci/prow/e2e-gcp-ovn ci/prow/e2e-vsphere-ovn
Just some e2e test failures

**Comment by sadasu** [comment]
/lgtm
Re-adding the lgtm that was lost after the previous rebase.

**Comment by sadasu** [comment]
/lgtm
Adding it back after it was lost after a rebase

**Pull request merged**